tests/tests_pytorch/utilities/test_model_helpers.py

```diff
@@ -71,6 +71,11 @@ def test_restricted_classmethod():
 
     RestrictedClass.restricted_cmethod()  # no exception
 
+    restricted_method = RestrictedClass().restricted_cmethod  # no exception
+
+    with pytest.raises(TypeError, match="cannot be called on an instance"):
+        restricted_method()
```

Comment:
These lines can directly replace the lines above. Might also want to throw in a `inspect.getmembers(RestrictedClass)  # no exception`

Reply:
Perfect!